DigitalLearningSolutions.Data/Models/DelegateUpload/DelegateTableRow.cs

```diff
@@ -0,0 +1,90 @@
+﻿namespace DigitalLearningSolutions.Data.Models.DelegateUpload
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using ClosedXML.Excel;
+
+    public enum RowStatus
+    {
+        NotYetProcessed,
+        Skipped,
+        Registered,
+        Updated
+    }
+
+    public class DelegateTableRow
+    {
+        public DelegateTableRow(IXLTable table, IXLRangeRow row)
+        {
+            string? FindFieldValue(string name)
+            {
+                var colNumber = table.FindColumn(col => col.FirstCell().Value.ToString() == name).ColumnNumber();
+                return row.Cell(colNumber).GetValue<string?>();
+            }
+
+            RowNumber = row.RowNumber();
+            CandidateNumber = FindFieldValue("DelegateID");
+            LastName = FindFieldValue("LastName");
+            FirstName = FindFieldValue("FirstName");
+            JobGroupId = int.TryParse(FindFieldValue("JobGroupID"), out var jobGroupId) ? jobGroupId : (int?)null;
+            Active = bool.TryParse(FindFieldValue("Active"), out var active) ? active : (bool?)null;
+            Answer1 = FindFieldValue("Answer1");
+            Answer2 = FindFieldValue("Answer2");
+            Answer3 = FindFieldValue("Answer3");
+            Answer4 = FindFieldValue("Answer4");
+            Answer5 = FindFieldValue("Answer5");
+            Answer6 = FindFieldValue("Answer6");
+            AliasId = FindFieldValue("AliasID");
+            Email = FindFieldValue("EmailAddress");
+            RowStatus = RowStatus.NotYetProcessed;
+        }
+
+        public int RowNumber { get; set; }
+        public string? CandidateNumber { get; set; }
+        public string? FirstName { get; set; }
+        public string? LastName { get; set; }
+        public int? JobGroupId { get; set; }
+        public bool? Active { get; set; }
+        public string? Answer1 { get; set; }
+        public string? Answer2 { get; set; }
+        public string? Answer3 { get; set; }
+        public string? Answer4 { get; set; }
+        public string? Answer5 { get; set; }
+        public string? Answer6 { get; set; }
+        public string? AliasId { get; set; }
+        public string? Email { get; set; }
+
+        public BulkUploadResult.ErrorReason? Error { get; set; }
+        public RowStatus RowStatus { get; set; }
+
+        public bool IsValid(IEnumerable<int> allowedJobGroupIds)

```

Comment:
It might be nicer to name this Validate() as IsValid sounds like a check with no state change? 🙏